server/src/main/java/keywhiz/service/daos/SecretSeriesMapper.java

```diff
@@ -20,26 +20,36 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import java.io.IOException;
 import java.util.Map;
+import java.util.Optional;
 import javax.inject.Inject;
 import keywhiz.api.ApiDate;
+import keywhiz.api.model.Group;
 import keywhiz.api.model.SecretSeries;
 import keywhiz.jooq.tables.records.SecretsRecord;
+import keywhiz.service.config.Readonly;
 import org.jooq.RecordMapper;
 
 public class SecretSeriesMapper implements RecordMapper<SecretsRecord, SecretSeries> {
   private static final TypeReference<Map<String, String>> MAP_STRING_STRING_TYPE =
       new TypeReference<>() {};
+
   private final ObjectMapper mapper;
+  private final GroupDAO groupDAO;
 
-  @Inject public SecretSeriesMapper(ObjectMapper mapper) {
+  @Inject public SecretSeriesMapper(
+      ObjectMapper mapper,
+      @Readonly GroupDAO groupDAO) {
     this.mapper = mapper;
+    this.groupDAO = groupDAO;
   }
 
   public SecretSeries map(SecretsRecord r) {
+    String owner = getOwner(r);
```

Comment:
Design-wise, this was the crux of the change: where and how should we perform the translation from the database's model of ownership, which is a long foreign key to the group table, to the API model of ownership which is the group's name? Fortunately, that's exactly the job of this class, at least on the read path. However, it did require injecting a dependency that we didn't have before, which was the original motivator for cleaning up the DI code in a previous change.

Reply:
Fantastic - otherwise, having to provide a group id for the owner (rather than the name) in the API would be confusing/painful. Nice work!